libr/crypto/p/crypto_punycode.c

```diff
@@ -0,0 +1,52 @@
+#include <r_util.h>
+#include <r_lib.h>
+#include <r_crypto.h>
+
+static int flag = 0;
+
+static int punycode_set_key(RCrypto *cry, const ut8 *key, int keylen, int mode, int direction) {
+	flag = direction;
+	return true;
+}
+
+static int punycode_get_key_size(RCrypto *cry) {
+	return 0;
+}
+
+static bool punycode_use(const char *algo) {
+	return !strcmp (algo, "punycode");
+}
+
+static int update(RCrypto *cry, const ut8 *buf, int len) {
+	ut8 *obuf = malloc (len + 100);
+	int olen;
+	if (flag) {
+		olen = r_punycode_decode (buf, obuf, len);
+	} else {
+		olen = r_punycode_encode (buf, obuf, len);
+	}
+	r_crypto_append (cry, obuf, olen);
```

Comment:
update should probably return bool instead of int.. and in case olen < len i would return 0/false, otherwise true